src/snowflake/snowpark/_internal/utils.py

```diff
@@ -1155,6 +1155,47 @@ def check_flatten_mode(mode: str) -> None:
         raise ValueError("mode must be one of ('OBJECT', 'ARRAY', 'BOTH')")
 
 
```

Comment:
Refactored from agg (dataframe.py and relational_grouped_dataframe.py) to move checks before AST generation.